Core/src/org/sleuthkit/autopsy/communications/AccountNode.java

```diff
@@ -47,25 +52,25 @@
      * @return The file name of the icon for the given Account Type.
      */
     final String getIconFileName(Account.Type type) {
-        if (type == Account.Type.CREDIT_CARD) {
+        if (type.equals(Account.Type.CREDIT_CARD)) {
             return "credit-card.png";
-        } else if (type == Account.Type.DEVICE) {
+        } else if (type.equals(Account.Type.DEVICE)) {
             return "image.png";
-        } else if (type == Account.Type.EMAIL) {
+        } else if (type.equals(Account.Type.EMAIL)) {
             return "email.png";
-        } else if (type == Account.Type.FACEBOOK) {
+        } else if (type.equals(Account.Type.FACEBOOK)) {
             return "facebook.png";
-        } else if (type == Account.Type.INSTAGRAM) {
+        } else if (type.equals(Account.Type.INSTAGRAM)) {
             return "instagram.png";
-        } else if (type == Account.Type.MESSAGING_APP) {
+        } else if (type.equals(Account.Type.MESSAGING_APP)) {
             return "messaging.png";
-        } else if (type == Account.Type.PHONE) {
+        } else if (type.equals(Account.Type.PHONE)) {
             return "phone.png";
-        } else if (type == Account.Type.TWITTER) {
+        } else if (type.equals(Account.Type.TWITTER)) {
             return "twitter.png";
-        } else if (type == Account.Type.WEBSITE) {
+        } else if (type.equals(Account.Type.WEBSITE)) {
             return "web-file.png";
-        } else if (type == Account.Type.WHATSAPP) {
+        } else if (type.equals(Account.Type.WHATSAPP)) {
```

Comment:
Why not use a switch statement here? I think it would improve readability significantly.

Reply:
Account.Type is not an enum or a primitive type.  

To allow for creation of new account types, and not having to separate out numeric ranges for predefined vs dynamically created account type,s we defined Account.Type as class.

Reply:
Got it, thanks.